core/src/main/java/com/linecorp/armeria/client/logging/LoggingClient.java

```diff
@@ -152,7 +153,7 @@ public LoggingClient(Client<I, O> delegate, LogLevel level) {
 
     @Override
     public O execute(ClientRequestContext ctx, I req) throws Exception {
-        if (sampler.isSampled()) {
+        if (sampler.isSampled(ctx.log())) {
```

Comment:
Is there any chance that a customized sampler uses the information from the `RequestLog` such as `ResponseHeaders`, etc.?
Then we should check this using listener.

Reply:
Wouldn't it increase the overhead because we are gonna have to add the listeners even if the request will not be logged actually?

Actually, this makes me think using `RequestLog` as an input for a `Sampler` doesn't make much sense. Should we just pass `RequestContext`?

Reply:
That's a good idea. Yeah, I agree that it's better to hand in `RequestContext` instead of `RequestLog`. 😄

Reply:
OK, then. @geminiKim, could you update the type parameters of `Sampler` from `RequestLog` to `ClientRequestContext` (for `LoggingClient`) and `ServiceRequestContext` (for `LoggingService`) ?

Reply:
Thank you for sharing your detailed discussion. Updated in 7977716.